### ISSUE TITLE
Installation issue: librsvg

### ISSUE THREAD

**Issue by omsai** [LLNL/spack#18865]
### Steps to reproduce the issue

```console
$ spack -d fetch -n librsvg
...
==> [2020-09-22-12:35:04.985115] Fetching from https://download.gnome.org/sources/librsvg2.44/librsvg-2.44.14.tar.xz failed.
==> [2020-09-22-12:35:04.985276] FetchError: All fetchers failed
==> [2020-09-22-12:35:04.985391] Error: All fetchers failed
```

The actual link is: https://download.gnome.org/sources/librsvg/2.44/librsvg-2.44.14.tar.xz

Clearly, the issue is a missing `/` between `librsvg` and `2.44`.  Fixed with:

```diff
diff --git a/var/spack/repos/builtin/packages/librsvg/package.py b/var/spack/repos/builtin/packages/librsvg/package.py
index c352255..4df5102 100644
--- a/var/spack/repos/builtin/packages/librsvg/package.py
+++ b/var/spack/repos/builtin/packages/librsvg/package.py
@@ -26,7 +26,7 @@ class Librsvg(AutotoolsPackage):
     depends_on('libxml2')

     def url_for_version(self, version):
-        url  = "https://download.gnome.org/sources/librsvg"
+        url  = "https://download.gnome.org/sources/librsvg/"
         url += "{0}/librsvg-{1}.tar.xz"
         return url.format(version.up_to(2), version)

```

PR incoming.

**Comment by omsai** [comment]
Ok the PR might require a bit more work because it fails to detect the installed `shared-mime-info@1.9`:

```
==> Installing librsvg
==> No binary for librsvg found: installing from source
==> Fetching https://download.gnome.org/sources/librsvg/2.44/librsvg-2.44.14.tar.xz
######################################################################## 100.0%
==> librsvg: Executing phase: 'autoreconf'
==> librsvg: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/tmp/pan14001admin/spack-stage/spack-stage-librsvg-2.44.14-3ttort7pjwu63qna7e5lvb6b4dsl73vs/spack-src/configure' '--prefix=/gpfs/gpfs1/apps2/librsvg/spack/opt/spack/linux-rhel6-westmere/gcc-9.2.0/librsvg-2.44.14-3ttort7pjwu63qna7e5lvb6b4dsl73vs'

1 error found in build log:
     97     checking for dlopen in -ldl... yes
     98     checking for cargo... cargo
     99     checking for rustc... rustc
     100    checking for some Win32 platform... no
     101    checking for native Win32... no
     102    checking for        cairo >= 1.15.12                cairo-png >= 1.15.12            freetype2 >= 20.0.14            gdk-pixbuf-2.0 >= 2.20  gio-2.0 >= 2.24.0                     glib-2.0 >= 2.4
            8.0                 libxml-2.0 >= 2.9.0             pangocairo >= 1.38.0            pangoft2 >= 1.38.0                    libcroco-0.6 >= 0.6.1... no
  >> 103    configure: error: Package requirements (    cairo >= 1.15.12                cairo-png >= 1.15.12            freetype2 >= 20.0.14            gdk-pixbuf-2.0 >= 2.20  gio-2.0 >= 2.24.0    
                        glib-2.0 >= 2.48.0              libxml-2.0 >= 2.9.0             pangocairo >= 1.38.0            pangoft2 >= 1.38.0                    libcroco-0.6 >= 0.6.1) were not met:
     104    
     105    Package 'shared-mime-info', required by 'gdk-pixbuf-2.0', not found
     106    
     107    Consider adjusting the PKG_CONFIG_PATH environment variable if you
     108    installed software in a non-standard prefix.
     109    

See build log for details:
  /tmp/pan14001admin/spack-stage/spack-stage-librsvg-2.44.14-3ttort7pjwu63qna7e5lvb6b4dsl73vs/spack-build-out.txt
```

```
$ spack concretize -f
==> Concretized librsvg                                                                                                       
 -   3ttort7  librsvg@2.44.14%gcc@9.2.0 arch=linux-rhel6-westmere                                                            
[+]  7z4j3qv      ^cairo@1.16.0%gcc@9.2.0~X+fc+ft~gobject~pdf~png~svg patches=7c4da77767fe9feb03f8051def0832f0c67f99162913275cfa127a88df19cf51 arch=linux-rhel6-westmere
[+]  ofeznm5          ^autoconf@2.69%gcc@9.2.0 arch=linux-rhel6-westmere                                                  
[+]  a6akg34          ^automake@1.16%gcc@9.2.0 arch=linux-rhel6-westmere                                                          
[+]  bkvucpw          ^fontconfig@2.13.1%gcc@9.2.0 arch=linux-rhel6-westmere                                            
[+]  dbaturg              ^font-util@1.3.2%gcc@9.2.0 fonts=encodings,font-adobe-100dpi,font-adobe-75dpi,font-adobe-utopia-100dpi,font-adobe-utopia-75dpi,font-adobe-utopia-type1,font-alias,fon
t-arabic-misc,font-bh-100dpi,font-bh-75dpi,font-bh-lucidatypewriter-100dpi,font-bh-lucidatypewriter-75dpi,font-bh-ttf,font-bh-type1,font-bitstream-100dpi,font-bitstream-75dpi,font-bitstream-s
peedo,font-bitstream-type1,font-cronyx-cyrillic,font-cursor-misc,font-daewoo-misc,font-dec-misc,font-ibm-type1,font-isas-misc,font-jis-misc,font-micro-misc,font-misc-cyrillic,font-misc-ethiop
ic,font-misc-meltho,font-misc-misc,font-mutt-misc,font-schumacher-misc,font-screen-cyrillic,font-sun-misc,font-winitzki-cyrillic,font-xfree86-type1 arch=linux-rhel6-westmere
[+]  ehqgmxf                  ^bdftopcf@1.0.5%gcc@9.2.0 arch=linux-rhel6-westmere                                                 
[+]  a3jjj22                      ^fontsproto@2.1.3%gcc@9.2.0 arch=linux-rhel6-westmere                                   
[+]  25wq5w5                          ^pkgconf@1.7.3%gcc@9.2.0 arch=linux-rhel6-westmere                                    
[+]  k7lgqby                          ^util-macros@1.19.1%gcc@9.2.0 arch=linux-rhel6-westmere                                             
[+]  gmnxx4l                      ^libxfont@1.5.2%gcc@9.2.0 arch=linux-rhel6-westmere                                     
[+]  cenzq34                          ^freetype@2.9.1%gcc@9.2.0 patches=08466355e8649235ff01f13b3e56bbd551c7cfb2ca97903cc11575c163ea32a3 arch=linux-rhel6-westmere
[+]  o3wygxq                          ^libfontenc@1.1.3%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  hhk7qgu                              ^xproto@7.0.31%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  sn566iq                              ^zlib@1.2.11%gcc@9.2.0+optimize+pic+shared arch=linux-rhel6-westmere
[+]  k3gfxil                          ^xtrans@1.3.5%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  aj7u5sh                  ^mkfontdir@1.0.7%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  5j7qfqw                      ^mkfontscale@1.1.2%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  zjmvufd              ^gperf@3.1%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  xwo4yrn              ^libuuid@1.0.3%gcc@9.2.0 arch=linux-rhel6-westmere                                                                                                                   
[+]  gse7dfs              ^libxml2@2.9.10%gcc@9.2.0~python arch=linux-rhel6-westmere
[+]  t65ykqb                  ^libiconv@1.16%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  oqigz3q                  ^xz@5.2.5%gcc@9.2.0~pic arch=linux-rhel6-westmere
[+]  muzzoud          ^glib@2.64.5%gcc@9.2.0~libmount patches=63e5a3efbdaf67cfb9707a20fd6e21e8826d1c785b20c6ec326f6d2b116950fc tracing=none arch=linux-rhel6-westmere
[+]  j4wilkf              ^gettext@0.21%gcc@9.2.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-rhel6-westmere
[+]  ojrdzwc                  ^bzip2@1.0.8%gcc@9.2.0+shared arch=linux-rhel6-westmere
[+]  qofaqbl                      ^diffutils@3.7%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  x7zs45k                  ^ncurses@6.2%gcc@9.2.0~symlinks+termlib arch=linux-rhel6-westmere
[+]  loqfq4e                  ^tar@1.32%gcc@9.2.0 arch=linux-rhel6-westmere                                                                                                                    
[+]  xiio4zw              ^libffi@3.3%gcc@9.2.0 patches=26f26c6f29a7ce9bf370ad3ab2610f99365b4bdd7b82e7c31df41a3370d685c0 arch=linux-rhel6-westmere
[+]  sqrdneg              ^meson@0.55.1%gcc@9.2.0 patches=45c35fce59361a994c49f8d3f5fd2595fc1b181ba4b5e219b7563752dd879a3b arch=linux-rhel6-westmere             
[+]  aqmhdrg                  ^ninja@1.10.1%gcc@9.2.0 arch=linux-rhel6-westmere                                                                                                                
[+]  vo3sht5                      ^python@3.8.5%gcc@9.2.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patc
hes=0d98e93189bc278fbc37a50ed7f183bd8aaf249a8e1670a465f0db6bb4f8cf87 arch=linux-rhel6-westmere                                                                                                 
[+]  2e4dx5r                          ^expat@2.2.9%gcc@9.2.0+libbsd arch=linux-rhel6-westmere
[+]  sscuh3e                              ^libbsd@0.10.0%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  eihckvv                          ^gdbm@1.18.1%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  gxnv24l                              ^readline@8.0%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  5mwavso                          ^openssl@1.1.1g%gcc@9.2.0+systemcerts arch=linux-rhel6-westmere
[+]  npdoavk                              ^perl@5.24.1%gcc@9.2.0+cpanm+shared+threads patches=0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac arch=linux-rhel6-westmere
[+]  cfzp5mq                          ^sqlite@3.33.0%gcc@9.2.0+column_metadata+fts~functions~rtree arch=linux-rhel6-westmere
[+]  du6tp7z                  ^py-setuptools@50.1.0%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  c2fad7z              ^pcre@8.44%gcc@9.2.0~jit+multibyte+utf arch=linux-rhel6-westmere
[+]  us32qls          ^libtool@2.4.6%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  z4mr7pe              ^m4@1.4.18%gcc@9.2.0+sigsegv patches=3877ab548f88597ab2327a2230ee048d2d07ace1062efe81fc92e91b7f39cd00,fc9b61654a3ba1a8d6cd78ce087e7c96366c290bc8d2c299f09828d793b853c
8 arch=linux-rhel6-westmere                                                                 
[+]  h37jpbh                  ^libsigsegv@2.12%gcc@9.2.0 arch=linux-rhel6-westmere     
[+]  hjbs6dc          ^pixman@0.40.0%gcc@9.2.0 arch=linux-rhel6-westmere          
[+]  i4kzjlh              ^libpng@1.6.37%gcc@9.2.0 arch=linux-rhel6-westmere            
[+]  otl44wf      ^gdk-pixbuf@2.40.0%gcc@9.2.0~x11 patches=6f6e6411051846f0e99e5f46e07c78428c4a2a88f51c202b3122972d91452f4c arch=linux-rhel6-westmere
[+]  mbzlh26          ^docbook-xsl@1.79.2%gcc@9.2.0 patches=a92c39715c54949ba9369add1809527b8f155b7e2a2b2e30cb4b39ee715f2e30 arch=linux-rhel6-westmere
[+]  a3bez5i              ^docbook-xml@4.5%gcc@9.2.0 arch=linux-rhel6-westmere      
[+]  3yqzcsk          ^gobject-introspection@1.56.1%gcc@9.2.0 patches=6f90bb267efa043ed70b900b4b8e2faf9e8133afae311893b01060356ea81bba arch=linux-rhel6-westmere
[+]  vq7msin              ^bison@2.4.1%gcc@9.2.0 arch=linux-rhel6-westmere                                                                                           
[+]  zku5hsf              ^flex@2.6.4%gcc@9.2.0+lex patches=09c22e5c6fef327d3e48eb23f0d610dcd3a35ab9207f12e0f875701c677978d3 arch=linux-rhel6-westmere
[+]  bnzzoul                  ^findutils@4.4.2%gcc@9.2.0 arch=linux-rhel6-westmere             
[+]  sxagonl                  ^help2man@1.47.11%gcc@9.2.0 arch=linux-rhel6-westmere
[+]  dg6t4v5          ^libjpeg-turbo@2.0.4%gcc@9.2.0 arch=linux-rhel6-westmere                                                                    
[+]  2zghakq              ^cmake@3.10.2%gcc@9.2.0~doc+ncurses+openssl+ownlibs~qt patches=1c540040c7e203dd8e27aa20345ecb07fe06570d56410a24a266ae570b1c4c39,dd3a40d4d92f6b2158b87d6fb354c277947c$
76424aa03f6dc8096cf3135f5d0 arch=linux-rhel6-westmere                          
[+]  ggvyoyk              ^nasm@2.14.02%gcc@9.2.0 arch=linux-rhel6-westmere                                                                                                                    [+]  kakf6qj          ^libtiff@4.1.0%gcc@9.2.0 arch=linux-rhel6-westmere                      
[+]  c2bfpi2          ^libxslt@1.1.29%gcc@9.2.0+crypto~python arch=linux-rhel6-westmere      
[+]  hjty4cd          ^shared-mime-info@1.9%gcc@9.2.0 arch=linux-rhel6-westmere             
[+]  bfoxy3o              ^intltool@0.51.0%gcc@9.2.0 patches=ca9d6562f29f06c64150f50369a24402b7aa01a3a0dc73dce55106f3224330a1 arch=linux-rhel6-westmere
[+]  hcb3vuv      ^libcroco@0.6.12%gcc@9.2.0 arch=linux-rhel6-westmere                     
[+]  mg3qoep      ^pango@1.41.0%gcc@9.2.0~X arch=linux-rhel6-westmere                                
[+]  f4rx6ci          ^harfbuzz@2.6.8%gcc@9.2.0~graphite2 arch=linux-rhel6-westmere                                                                                                     
[+]  ysysssr              ^icu4c@67.1%gcc@9.2.0 cxxstd=11 arch=linux-rhel6-westmere                                         
[+]  q6gfsll      ^rust@1.46.0%gcc@9.2.0+analysis+clippy~rls+rustfmt+src extra_targets=none arch=linux-rhel6-westmere
```

**Comment by omsai** [comment]
After adding the dependency it still fails with some rust related error:

```console
$ spack install
...
==> Installing librsvg                                      
==> No binary for librsvg found: installing from source            
==> Using cached archive: /gpfs/gpfs1/apps2/librsvg/spack/var/spack/cache/_source-cache/archive/6a/6a85a7868639cdd4aa064245cc8e9d864dad8b8e9a4a8031bb09a4796bc4e303.tar.xz
==> librsvg: Executing phase: 'autoreconf'                                                         
==> librsvg: Executing phase: 'configure'
==> librsvg: Executing phase: 'build'                             
==> Error: ProcessError: Command exited with status 2:      
    'make' '-j12'                                                  
                                  
6 errors found in build log:                                                                                         
     317        |                 ---------------------------^^^^-------------------
     318        |                 |                          |                                                                          
     319        |                 |                          immutable borrow occurs here
     320        |                 mutable borrow occurs here
     321        |                 mutable borrow later used here                                               
     322                           
  >> 323    error: aborting due to previous error
     324                                                                                                                  
     325    For more information about this error, try `rustc --explain E0502`.
  >> 326    error: could not compile `nalgebra`.                                                                      
     327                     
     328    To learn more, run the command again with --verbose.
     329    warning: build failed, waiting for other jobs to finish...
  >> 330    error: build failed                    
  >> 331    make[2]: *** [/tmp/pan14001admin/spack-stage/spack-stage-librsvg-2.44.14-tnansiakofc7avllbaa57vav5nfwfuyu/spack-src/target/release/librsvg_internals.a] Error 101
     332    make[2]: Leaving directory `/tmp/pan14001admin/spack-stage/spack-stage-librsvg-2.44.14-tnansiakofc7avllbaa57vav5nfwfuyu/spack-src'
  >> 333    make[1]: *** [all-recursive] Error 1
     334    make[1]: Leaving directory `/tmp/pan14001admin/spack-stage/spack-stage-librsvg-2.44.14-tnansiakofc7avllbaa57vav5nfwfuyu/spack-src'
  >> 335    make: *** [all] Error 2
                                        
See build log for details:                                  
  /tmp/pan14001admin/spack-stage/spack-stage-librsvg-2.44.14-tnansiakofc7avllbaa57vav5nfwfuyu/spack-build-out.txt
```

```diff
diff --git a/var/spack/repos/builtin/packages/librsvg/package.py b/var/spack/repos/builtin/packages/librsvg/package.py
index c352255..8c86d25 100644
--- a/var/spack/repos/builtin/packages/librsvg/package.py
+++ b/var/spack/repos/builtin/packages/librsvg/package.py
@@ -24,9 +24,10 @@ class Librsvg(AutotoolsPackage):
     depends_on("pango")
     depends_on('libffi')
     depends_on('libxml2')
+    depends_on('shared-mime-info')
 
     def url_for_version(self, version):
-        url  = "https://download.gnome.org/sources/librsvg"
+        url  = "https://download.gnome.org/sources/librsvg/"
         url += "{0}/librsvg-{1}.tar.xz"
         return url.format(version.up_to(2), version)
```

Hmmm... maybe I should try compiling the current stable librsvg 2.50.0 to avail of help from upstream.
(edit s/rsvg/librsvg/)

**Comment by omsai** [comment]
The latest librsvg@2.50.0 requires the cairo+gobject variant whereas the current librsvg only uses vanilla cairo:

```console
$ spack install
...
==> Installing librsvg                                                                                  
==> No binary for librsvg found: installing from source                                                                                                
==> Warning: There is no checksum on file to fetch librsvg@2.50.0 safely.                                                                                
==>   Fetch anyway? [y/N] y                                                                                                                                   
==> Fetching https://download.gnome.org/sources/librsvg/2.50/librsvg-2.50.0.tar.xz                                                                     
######################################################################## 100.0%                                                                          
==> librsvg: Executing phase: 'autoreconf'                                                                                                                    
==> librsvg: Executing phase: 'configure'                                                                                                              
==> Error: ProcessError: Command exited with status 1:                                                                                                   
    '/tmp/pan14001admin/spack-stage/spack-stage-librsvg-2.50.0-dyp747b2vrkms6szgmeuqpbabrq5dzxc/spack-src/configure' '--prefix=/gpfs/gpfs1/apps2/librsvg/spack/opt/spack/linux-rhel6-westmere/g
cc-9.2.0/librsvg-2.50.0-dyp747b2vrkms6szgmeuqpbabrq5dzxc'                                                                                              
                                                                                                                                                         
1 error found in build log:                                                                                                                                   
     120    checking for dlopen in -ldl... yes                                                                        
     121    checking for cargo... cargo                                                                                         
     122    checking for rustc... rustc                                                                                                                                                        
     123    checking for some Win32 platform... no                                                                        
     124    checking for native Win32... no                                                                                                                                                    
     125    checking for LIBRSVG... no                                                                             
  >> 126    configure: error: Package requirements (    cairo >= 1.16.0                 cairo-png >= 1.16.0             cairo-gobject >= 1.16.0         freetype2 >= 20.0.14            gdk-pix
buf-2.0 >                                                                                                                
            = 2.20      gio-2.0 >= 2.24.0                       glib-2.0 >= 2.50.0              harfbuzz >= 2.0.0               libxml-2.0 >= 2.9.0             pangocairo >= 1.38.0          p
angoft2 >= 1.38.0                                                                         
                       ) were not met:
     127
     128    Package 'cairo-gobject', required by 'virtual:world', not found
     129
     130    Consider adjusting the PKG_CONFIG_PATH environment variable if you
     131    installed software in a non-standard prefix.
     132

See build log for details:
  /tmp/pan14001admin/spack-stage/spack-stage-librsvg-2.50.0-dyp747b2vrkms6szgmeuqpbabrq5dzxc/spack-build-out.txt
```

Configure succeeds with these changes:

```diff
diff --git a/var/spack/repos/builtin/packages/librsvg/package.py b/var/spack/repos/builtin/packages/librsvg/package.py
index c352255..942b183 100644
--- a/var/spack/repos/builtin/packages/librsvg/package.py
+++ b/var/spack/repos/builtin/packages/librsvg/package.py
@@ -12,21 +12,22 @@ class Librsvg(AutotoolsPackage):
     homepage = "https://wiki.gnome.org/Projects/LibRsvg"
     url      = "https://download.gnome.org/sources/librsvg/2.44/librsvg-2.44.14.tar.xz"

-    version('2.44.14', sha256='6a85a7868639cdd4aa064245cc8e9d864dad8b8e9a4a8031bb09a4796bc4e303')
+    version('2.50.0', sha256='b3fadba240f09b9c9898ab20cb7311467243e607cf8f928b7c5f842474ee3df4')

     depends_on("gobject-introspection", type='build')
     depends_on("pkgconfig", type='build')
     depends_on("rust", type='build')
-    depends_on("cairo")
+    depends_on("cairo+gobject")
     depends_on("gdk-pixbuf")
     depends_on("glib")
     depends_on("libcroco")
     depends_on("pango")
     depends_on('libffi')
     depends_on('libxml2')
+    depends_on('shared-mime-info')

     def url_for_version(self, version):
-        url  = "https://download.gnome.org/sources/librsvg"
+        url  = "https://download.gnome.org/sources/librsvg/"
         url += "{0}/librsvg-{1}.tar.xz"
         return url.format(version.up_to(2), version)

```

**Comment by adamjstewart** [comment]
I'm guessing this was fixed by #18869
